metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/AggregationFieldType.java

```diff
@@ -0,0 +1,72 @@
+package eu.europeana.enrichment.api.internal;
+
+import eu.europeana.enrichment.utils.EntityType;
+import eu.europeana.metis.schema.jibx.Aggregation;
+import eu.europeana.metis.schema.jibx.ResourceOrLiteralType;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+public enum AggregationFieldType {
```

Comment:
I'm sure that if we merge this with FieldType (and make RDF the type of the method input istead of ProxyType or Aggregation) we can actually make code a lot more generic and reduce code duplication all around.